### ISSUE TITLE
Column Wrapping For Wide Tables

### ISSUE THREAD

**Issue by greenbergjosh** [olifolkerd/tabulator#1485]
**Is your feature request related to a problem? Please describe.**
I have a very wide table and would like to be able to display all of the data at once within the screen width without horizontal scrolling and without a custom RowFormatter. I do not want to use the RowFormatter because it eliminates column headers which are the intuitive location for column sorting.

**Describe the solution you'd like**
I would like the columns to wrap. This is not wrapping of content within a cell, but rather, if the number of columns cannot be displayed in the visible (or declared) table width, then the columns (and the associated data for those columns in each corresponding row) should wrap.

--------------------------------------
Col1          Col2           Col3          Col4          this is the header - columns wrapped
        Col5                 Col6               Col7         
--------------------------------------
d11          d12              d13            d14          this is a row of data - columns wrapped same as header
       d15                    d16                  d17        
-  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -
d21          d22              d23            d24          this is another row of data - columns wrapped the same
       d25                    d26                  d27         

In short, I would like multi-line headers and associated multi-line rows. As the browser width decreases, the number of columns per line would decrease, and consequently, the number of lines-per-row would eventually increase.  

**Describe alternatives you've considered**
I have considered the RowFormatter and nested tables/grids..  However, I only want a single header, with all columns, and the normal functionality of column headers (e.g. sorting)

**Additional context**
Please see ascii drawing above.  Forgive me that it looks better in ASCII than in the rendered form of the question.  I hope it is still clear that each row and the header are using multiple lines.

Thank you for considering my request.

**Comment by olifolkerd** [comment]
Hey @greenbergjosh 

The good news is this is already a feature, have a look at the [Responsive Column Layout](http://tabulator.info/docs/4.0/layout#responsive-collapse) documentation, the collapse mode will show data from columns that do not fit on the screen as a list under the rows that do fit.

You can also use a custom list formatter to display the missing rows in a different fashion if you like.

Cheers

Oli :)

**Comment by greenbergjosh** [comment]
Hello Oli,

Thank you for the reply.  I played a bit with the Responsive Column Layout, but it is really not quite the same thing.  A few points will clarify the difference.

1. Using responsive column layout, even with a custom list formatter, the column headers are still removed from the top of the table.  It is the column headers that a user would typically click on to do sorting, and also what the user would view to identify the semantics of the data by its position.
2. In the absence of column headers the semantics is recovered by redundantly storing the column name along with the data in every row.  This significantly increases the amount of data being displayed without providing any additional value (since it is redundant in every row).
3. The column headers could also be clicked to highlight the corresponding cells. Consider a table that has wrapped such that it requires three lines to display all of the column headers (and therefore three lines, per row, to display the data).  By allowing a column header to be clicked, and highlighting the cells corresponding to that header, it would be easy for the user to see which data falls under that wrapped column.  In the absence of column headers, this is not possible.
4. Returning to sorting.  In the absence of column headers, the only reasonable places to store sorting options for the listed columns (as opposed to the remaining column headers) would be on the redundant names that appear in each row.  This means every name formatted by the list formatter to take the place of a column header must be a link that calls the sort function for that listed column.  Your API certainly supports this easily, but it doesn't naturally support the toggle (asc, desc), so that state must be maintained elsewhere.  The column headers toggle between asc, desc naturally.

I recognize that the resulting table may appear a bit busy, but it provides the minimal context and content for displaying all data, and item 3 above allows the user a simple gesture that clarifies her location.  It also leverages existing column header functionality rather than requiring further work to add sorting, filtering, etc. back in.

**Comment by olifolkerd** [comment]
Hey @greenbergjosh 

Thanks for your detailed explanation,

While i do take your points on board about sorting etc, i simply feel that at the point the UI starts wrapping column headers and cells it makes for an extremely untidy and unintuitive UI.

At this stage i will not be looking to add that as an option to the responsive layout system, but if you feel it is something that would benefit your implementation you are of course welcome to add the mode yourself and submit a pull request.

Thanks for your suggestion.

Cheers

Oli :)
